app/scripts/migrations/048.js

```diff
@@ -24,7 +25,8 @@ function transformState (state = {}) {
   delete state.NetworkController?.settings
 
   // 2. Delete NetworkController.provider or rename rpcTarget key
-  if (state.NetworkController?.provider?.type === 'rpc') {
+  const providerType = state.NetworkController?.provider?.type
+  if (providerType === 'rpc' || providerType === 'localhost') {
```

Comment:
Will this bump users of a custom RPC onto main-net? If so, I could imagine users being surprised that what they thought was test-ether or something else is now main-net ether. We do show visual indications of the current network, so this is "theoretically safe", but I could imagine problems.

Reply:
Yeah, it does. We could switch the user to a test network instead of Mainnet in the migration, or add another alert.